crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -178,11 +192,37 @@ where
     ) -> Result<(), EnvironmentError2> {
         debug!("building project environment at {}", self.path.display());
         let manifest_path = self.manifest_path();
-        let lockfile_path = {
-            // TODO: generate a lockfile with pkgdb
-            manifest_path.parent().unwrap().join("dummy_lockfile.json")
+        let lockfile_path = self.lockfile_path();
+        let maybe_lockfile = if lockfile_path.exists() {
+            debug!("found existing lockfile: {}", lockfile_path.display());
+            // TODO: not yet implemented in pkgdb
+            // Some(lockfile_path.as_ref())
+            None
+        } else {
+            debug!("no existing lockfile found");
+            None
         };
-        debug!("generated lockfile: {}", lockfile_path.display());
+        let mut lockfile_json = resolve_manifest_to_lockfile(
+            Path::new(&PKGDB_BIN.as_str()),
+            &manifest_path,
+            maybe_lockfile,
+        )?;
+        if let Ok::<PkgDbError, _>(pkgdb_err) = serde_json::from_value(lockfile_json.clone()) {
+            return Err(EnvironmentError2::FailedResolution(pkgdb_err));
+        }
+        // TODO: pkgdb needs to add a `url` field, until that's done we do it manually
+        let nixpkgs_url = Value::String(
+            "github:NixOS/nixpkgs/e8039594435c68eb4f780f3e9bf3972a7399c4b1".to_string(),
+        );
+        lockfile_json["registry"]["inputs"]["nixpkgs"]["url"] = nixpkgs_url.clone();
+        if let Value::Object(ref mut packages) = lockfile_json["packages"] {
+            for (_p_name, p_value) in packages.iter_mut() {
+                p_value[system]["url"] = nixpkgs_url.clone();
+            }
+        }
```

Comment:
note: wonder if we could sneak in a lock for `mkEnv` here 🤔

Reply:
I think that makes sense in general, but not here because this is a hack that should be removed with the next set of updates to `pkgdb`